src/file-table-engine/src/manifest/immutable.rs

```diff
@@ -0,0 +1,88 @@
+// Copyright 2023 Greptime Team
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use object_store::ObjectStore;
+use serde::{Deserialize, Serialize};
+use snafu::{ensure, ResultExt};
+use table::metadata::RawTableInfo;
+
+use crate::error::{
+    CheckObjectSnafu, DecodeJsonSnafu, DeleteTableManifestSnafu, EncodeJsonSnafu,
+    ReadTableManifestSnafu, Result, WriteImmutableManifestSnafu, WriteTableManifestSnafu,
+};
+const IMMUTABLE_MANIFEST_FILE: &str = "_immutable_manifest";
+
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, Eq)]
+pub struct ImmutableMetadata {
+    pub table_info: RawTableInfo,
```

Comment:
Adds a `version` field to represent the serialization protocol version for future compatibility.